### PR TITLE
Added Device Management Functionality

### PR DESCRIPTION
**Pull request by caffeinatedMike**
Closes Issue #16  
 - Refactored a stale Pull Request #26 to work with current code base
 - Updated README.md to include the new publicly available function
 - Updated README.md's Supported Device list to officially support my (older) router model

Co-authored-by gskluzacek@gmail.com

### CODE REVIEW

**Comment by MatMaul** [comment]
Nice 🙂 I don't know if I'll have time I am going to holidays. Can you squash your commits into one with rebase ?
Also I think it would be nice to add @gskluzacek as a co author, see here:
https://help.github.com/articles/creating-a-commit-with-multiple-authors

**Comment by caffeinatedMike** [comment]
Sure, I'll work on doing that in the next few hours

**Comment by MatMaul** [comment]
Ok I'll do it 😉

**Comment by caffeinatedMike** [comment]
@MatMaul I'm happy to say the new feature is fully operational & returning expected results. I believe this feature is ready for the master branch whenever you have the time to review it this weekend :grin: :tada:

File: pynetgear/__init__.py [hunk 1]
```diff
@@ -271,6 +272,71 @@ def tofloats(lst): return (float(t) for t in lst)
...
+        """
+        Start a configuration session.
+        For managing router admin functionality (ie allowing/blocking devices)
+        """
+        _LOGGER.info("Config start")
+
+        message = "<NewSessionID>{session_id}</NewSessionID>".format(session_id=SESSION_ID)
+
+        success, _ = self._make_request(
+            SERVICE_DEVICE_CONFIG, "ConfigurationStarted", message)
```
Comment (MatMaul) [inline pynetgear/__init__.py]: _make_request can take a dict as params that will be inserted as a body. Here for example you can directly pass `{"NewSessionID": SESSION_ID}` like here:
https://github.com/MatMaul/pynetgear/blob/master/pynetgear/__init__.py#L93

This applies everywhere applicable :)

File: pynetgear/__init__.py [hunk 2]
```diff
@@ -271,6 +272,71 @@ def tofloats(lst): return (float(t) for t in lst)
...
+
+        message = "<NewStatus>ChangesApplied</NewStatus>"
+
+        success, _ = self._make_request(
+            SERVICE_DEVICE_CONFIG, "ConfigurationFinished", message)
+        
+        self.config_started = not success
+        return success
+    
+    def allow_block_device(self, mac_addr, device_status='Block'):
```
Comment (MatMaul) [inline pynetgear/__init__.py]: for device_status please define 2 constants BLOCK and ALLOW so users can use them too directly.

Comment (caffeinatedMike) [reply pynetgear/__init__.py]: Fixed :)

File: pynetgear/__init__.py (hunk 1 above)
Comment (caffeinatedMike) [reply pynetgear/__init__.py]: Fixed :)

**Comment by MatMaul** [comment]
Final version here, I'll merge soon.
https://github.com/MatMaul/pynetgear/tree/allow-device-blocking

Next time please use pep8 for the syntax check, a lot of empty lines with spaces or tabs instead of spaces :)
Ignore `E501 line too long` however if the length is still reasonable, 80 is too low.

**Comment by caffeinatedMike** [comment]
That's what I figured. I'll look into the new concept as well. But, regardless I NEED a way to use this functionality in my automations. It's perfect for so many cases of mine. The best one the comes to mind is:

1. New Device detected on router
2. Send Notification to my Android phone with 2 buttons (Allow/Deny)
3. If Deny clicked HA would block access to the new device/intruder

**Comment by MatMaul** [comment]
https://www.home-assistant.io/blog/2018/09/28/release-79/
https://github.com/home-assistant/architecture/issues/36

Not sure yet if it's useful. Doesn't look like there is any way to act on a device. However we could probably have an option to create a switch and create a device containing the device_tracker and the switch.

**Comment by MatMaul** [comment]
Tested approved and merged, thanks :)
Probably possible, to be seen !

**Comment by caffeinatedMike** [comment]
@MatMaul Glad I could help! Hit me up when you need more help, I enjoy being able to contribute :)

And let's get on that, incorporating this block functionality into Home Assistant somehow, ASAP ;)

**Comment by MatMaul** [comment]
New version released, PR sent on home-assistant.
https://github.com/home-assistant/home-assistant/pull/17652

**Pull request closed without merging**